tests/test_components/test_heat_charge.py

```diff
@@ -10,6 +10,84 @@
 from ..utils import assert_log_level
 
 
+class CHARGE_SIMULATION:
```

Comment:
Hmm I see the motivation for this, but wonder if it'd be more scalable with pytest fixtures as enables better erroring. I say this because maybe you want to test components individually and say, if there's a validation error or the declaration goes wrong, it's better than to have it all contained in a class like this. There's a few ways to implement fixtures but ultimately you want to check modular declaration in a parametric setting

Reply:
I actually created this class as a placeholder for variables rather than for testing (in C++ I would have created a struct instead). I do see, however, how this can be confusing, specially because I have stuff that aren't variables (like monitors). I will refactor this so that objects that might raise error are placed within the test function.
Thanks Dario!

Reply:
Nice, sounds good! Maybe something on these lines might be helpful: https://docs.pytest.org/en/6.2.x/fixture.html

Reply:
This is very interesting, actually. I have refactored this test though I'm not sure I'm making the most of fixtures. Let's discuss once I push changes